<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>V3DClass</class>
 <widget class="QWidget" name="V3DClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>556</width>
    <height>745</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>556</width>
    <height>745</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>V3D</string>
  </property>
  <property name="toolTip">
   <string/>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="light3D" name="light" native="true">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>90</y>
     <width>301</width>
     <height>281</height>
    </rect>
   </property>
   <property name="focusPolicy">
    <enum>Qt::StrongFocus</enum>
   </property>
  </widget>
  <widget class="QLabel" name="Fps">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>50</y>
     <width>171</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>微软雅黑</family>
     <pointsize>12</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">color: rgb(0, 0, 0);
</string>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QPushButton" name="readUltrButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>81</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>HeadFile</string>
   </property>
  </widget>
  <widget class="QPushButton" name="readCTButton">
   <property name="geometry">
    <rect>
     <x>100</x>
     <y>10</y>
     <width>81</width>
     <height>31</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <property name="text">
    <string>NoHead</string>
   </property>
  </widget>
  <widget class="QGroupBox" name="groupBox_8">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>380</y>
     <width>301</width>
     <height>71</height>
    </rect>
   </property>
   <property name="title">
    <string>Gamma</string>
   </property>
   <widget class="QLabel" name="Label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Gamma Correction</string>
    </property>
   </widget>
   <widget class="QSlider" name="gammaCorrectionSlider">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>251</width>
      <height>19</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>26</number>
    </property>
    <property name="singleStep">
     <number>1</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="value">
     <number>16</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="gammaCorrectionL">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>40</y>
      <width>21</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>1.6</string>
    </property>
   </widget>
  </widget>
  <widget class="QPushButton" name="TF">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>10</y>
     <width>41</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>TF</string>
   </property>
  </widget>
  <widget class="QScrollArea" name="scrollArea">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>510</y>
     <width>531</width>
     <height>231</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="styleSheet">
    <string notr="true">QScrollArea{border:0px solid #555;}</string>
   </property>
   <property name="widgetResizable">
    <bool>true</bool>
   </property>
   <widget class="QWidget" name="scrollAreaWidgetContents">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>514</width>
      <height>700</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>700</height>
     </size>
    </property>
    <widget class="QGroupBox" name="groupBox_6">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>511</width>
       <height>111</height>
      </rect>
     </property>
     <property name="title">
      <string>Photon Mapping</string>
     </property>
     <widget class="QLabel" name="photonNumberL">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>191</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>photon number 4000000</string>
      </property>
     </widget>
     <widget class="QSlider" name="photonNumberSlider">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>491</width>
        <height>19</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="minimum">
       <number>50</number>
      </property>
      <property name="maximum">
       <number>1500</number>
      </property>
      <property name="singleStep">
       <number>10</number>
      </property>
      <property name="pageStep">
       <number>10</number>
      </property>
      <property name="value">
       <number>400</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSlider" name="photonExposureSlider">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>491</width>
        <height>19</height>
       </rect>
      </property>
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>200</number>
      </property>
      <property name="singleStep">
       <number>1</number>
      </property>
      <property name="pageStep">
       <number>1</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="photonExposureL">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>60</y>
        <width>31</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="photonNumberL_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>191</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>photon mapping HDR Exposure</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>140</y>
       <width>511</width>
       <height>121</height>
      </rect>
     </property>
     <property name="title">
      <string>Sampling</string>
     </property>
     <widget class="QSlider" name="samplingCount">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>491</width>
        <height>19</height>
       </rect>
      </property>
      <property name="minimum">
       <number>100</number>
      </property>
      <property name="maximum">
       <number>3000</number>
      </property>
      <property name="singleStep">
       <number>1</number>
      </property>
      <property name="value">
       <number>1000</number>
      </property>
      <property name="sliderPosition">
       <number>1000</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="sampleCL">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>131</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Sample Count  1000</string>
      </property>
     </widget>
     <widget class="QSlider" name="gradientCount">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>491</width>
        <height>19</height>
       </rect>
      </property>
      <property name="minimum">
       <number>100</number>
      </property>
      <property name="maximum">
       <number>3000</number>
      </property>
      <property name="singleStep">
       <number>1</number>
      </property>
      <property name="value">
       <number>1000</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="gradientCL">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>131</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Gradient Count 1000</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>270</y>
       <width>511</width>
       <height>211</height>
      </rect>
     </property>
     <property name="title">
      <string>lighting Source</string>
     </property>
     <widget class="QLabel" name="ambiemtIL">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>191</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Ambient Instenity   0.0</string>
      </property>
     </widget>
     <widget class="QSlider" name="ambientSlider">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>431</width>
        <height>19</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="singleStep">
       <number>1</number>
      </property>
      <property name="pageStep">
       <number>10</number>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QPushButton" name="ambientButton">
      <property name="geometry">
       <rect>
        <x>450</x>
        <y>50</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Color</string>
      </property>
     </widget>
     <widget class="QLabel" name="diffuseIL">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>171</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Diffuse Instenity   1.0</string>
      </property>
     </widget>
     <widget class="QSlider" name="diffuseSlider">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>431</width>
        <height>19</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QPushButton" name="diffuseButton">
      <property name="geometry">
       <rect>
        <x>450</x>
        <y>90</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Color</string>
      </property>
     </widget>
     <widget class="QLabel" name="specularIL">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>110</y>
        <width>181</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Specular Instenity 0.0</string>
      </property>
     </widget>
     <widget class="QSlider" name="specularSlider">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>431</width>
        <height>19</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QPushButton" name="specularButton">
      <property name="geometry">
       <rect>
        <x>450</x>
        <y>130</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Color</string>
      </property>
     </widget>
     <widget class="QLabel" name="distanceL">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>150</y>
        <width>141</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Distance  2.5</string>
      </property>
     </widget>
     <widget class="QSlider" name="distanceSlider">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>170</y>
        <width>381</width>
        <height>19</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>25</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QGroupBox" name="groupBox_5">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>460</y>
     <width>521</width>
     <height>51</height>
    </rect>
   </property>
   <property name="title">
    <string> HDR exposure</string>
   </property>
   <widget class="QSlider" name="exposureSlider">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>20</y>
      <width>441</width>
      <height>20</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>50</number>
    </property>
    <property name="singleStep">
     <number>2</number>
    </property>
    <property name="pageStep">
     <number>2</number>
    </property>
    <property name="value">
     <number>25</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="exposureL">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>20</y>
      <width>21</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>2.5</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="HDRcheckBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="groupBox_4">
   <property name="geometry">
    <rect>
     <x>320</x>
     <y>90</y>
     <width>221</width>
     <height>361</height>
    </rect>
   </property>
   <property name="title">
    <string>Illumination Choosen</string>
   </property>
   <widget class="QRadioButton" name="localLightCheck">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>181</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Local Illumination(Phong)</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <property name="autoRepeat">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="noLightCheck">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Not Illumination</string>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="directIlluminationCheck">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>181</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Direct Illumination</string>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="inDirectIlluminationCheck">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>110</y>
      <width>191</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Photon InDirect Illumination</string>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="DirectIndirectPhongCheck">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>140</y>
      <width>201</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Direct + InDirect Illumination</string>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="Label_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>210</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Light Attenuation</string>
    </property>
   </widget>
   <widget class="QLabel" name="lightAttenuationL">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>240</y>
      <width>21</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>0.2</string>
    </property>
   </widget>
   <widget class="QSlider" name="lightAttenuationSlider">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>240</y>
      <width>181</width>
      <height>19</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>40</number>
    </property>
    <property name="singleStep">
     <number>1</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSlider" name="surfaceShadingParameter">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>190</y>
      <width>181</width>
      <height>20</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="singleStep">
     <number>10</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="SSPL">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>190</y>
      <width>21</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>1.0</string>
    </property>
   </widget>
   <widget class="QLabel" name="Label_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>170</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Surface parameter</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_7">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>280</y>
      <width>221</width>
      <height>91</height>
     </rect>
    </property>
    <property name="title">
     <string>Skin Model</string>
    </property>
    <widget class="QLabel" name="colorBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>201</width>
       <height>21</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">image: url(:/V3D/Resources/colobox.png);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QSlider" name="colorBoxSlider">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>201</width>
       <height>19</height>
      </rect>
     </property>
     <property name="maximum">
      <number>200</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="value">
      <number>100</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="colorBoxL">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>0.0</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QPushButton" name="VIS">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>50</y>
     <width>41</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>VIS</string>
   </property>
  </widget>
  <widget class="QPushButton" name="loadTF">
   <property name="geometry">
    <rect>
     <x>390</x>
     <y>10</y>
     <width>71</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>Load TF</string>
   </property>
  </widget>
  <widget class="QPushButton" name="saveTF">
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>10</y>
     <width>71</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>Save TF</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="useTFColors">
   <property name="geometry">
    <rect>
     <x>340</x>
     <y>50</y>
     <width>101</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>Use TF Color</string>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>light3D</class>
   <extends>QWidget</extends>
   <header>light3d.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="v3d.qrc"/>
 </resources>
 <connections/>
</ui>
